transient/config.py

```diff
@@ -0,0 +1,261 @@
+"""Supports the creation and validation of Transient-run configurations
+"""
+
+import toml
+import sys
+
+from marshmallow import Schema, fields, post_load, ValidationError
+
+from typing import Any, Dict, MutableMapping
+
+
+default_config_file = """# A Transient config file
+
+#
+# Transient arguments are set in a similar manner compared to the command line.
+# A few differences exist:
+#
+#   1. All values are set using an equal sign:
+#
+#          shutdown-timeout = 20
+#
+#
+#   2. Strings are encapsulated with double quotes
+#
+#          name = "vm_name"
+#
+#
+#   3. Flags are set to Booleans:
+#
+#          prepare-only = true
+#          ssh-console = false
+#
+#
+#   4. Lists are encapsulated by square brackets:
+#
+#          shared-folder = [
+#              "/path/on/host:/path/on/guest",
+#              "/path/on/host2:/path/on/guest2"
+#          ]
+#
+
+[transient]
+
+copy-in-before   = []
+copy-out-after   = []
+copy-timeout     = 0
+image            = ["centos/7:2004.01"]
+image_backend    = ""
+image_frontend   = ""
+name             = "vm_name"
+prepare-only     = false
+qmp-timeout      = 10
+shared-folder    = []
+shutdown-timeout = 20
+ssh-bin-name     = ""
+ssh-command      = ""
+ssh-console      = false
+ssh-port         = 22
+ssh-user         = "vagrant"
+ssh-with-serial  = false
+
+
+#
+# QEMU arguments are set in a similar manner compared to the command line.
+#
+# The only difference is that the QEMU argument string is split by whitespace
+# and encapsulated in a list.
+#
+
+[qemu]
+
+qemu-args = [
+    "-nographic",
+    "-enable-kvm",
+    "-m", "1G"
+]
+"""
+
+
+class Config(Dict[Any, Any]):
+    """Creates an argument dictionary that allows dot notation to access values
+
+    Example:
+
+        >>> args = Config({'arg1': 1, 'arg2': 2})
+        >>> args['arg1'] == args.arg1
+
+    """
+
+    def __init__(self, *args: Dict[Any, Any], **kwargs: Dict[Any, Any]) -> None:
+        super(Config, self).__init__(*args, **kwargs)
+        for arg in args:
+            if isinstance(arg, dict):
```

Comment:
We just ignore any non-dict arg?

Reply:
Yes. The intent of this class is to take a dictionary and give it the ability to use dot notation for accessing values.

On second thought, it may be better to use `args: List[Dict[Any, Any]]` in the parameter.

Reply:
Resolved by setting the parameter to `args: List[Dict[Any, Any]]`